src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableHashSet_1.cs

```diff
@@ -639,7 +639,7 @@ private static bool IsSupersetOf(IEnumerable<T> other, MutationInput origin)
         private static MutationResult Add(T item, MutationInput origin)
         {
             OperationResult result;
-            int hashCode = origin.EqualityComparer.GetHashCode(item);
+            int hashCode = origin.EqualityComparer.GetHashCode(item!);
```

Comment:
But now it seems like that would actually not just be for TryGetValue, but really all operations on the hash set, at which point we'd be better off using a notnull constraint?  Maybe instead of changing the annotations, open an issue about it and keep the ! changes you have here.

Reply:
(And actually, the right answer is probably to do a null check prior to calling GetHashCode, as we do in `HashSet<T>`.)